### PR TITLE
J setup game screens

### PR DESCRIPTION
**Pull request by JermyTan**
1. Add Game map (can be dragged around)
2. Add mock level data
3. Add stage overview and avatar selector
4. Add `Stage` and `Avatar` model

TODO:
1. Link selected stage to gameplay
2. Modify `Stage` to contain enemies data
3. Add power ups for holy knight

### CODE REVIEW

File: GameOfRunes/GameOfRunes/Models/Stage.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+    
+    let name: String
+    let groupName: String
+    let category: Category
+    let relativePositionRatioInMap: (x: CGFloat, y: CGFloat)
+    let arena: ArenaType
+    let difficulty: Int
+    let numWaves: Int
+    // might want to add the array of enemies here? But how do we determine the numWaves?
```
Comment (dongsiji) [inline GameOfRunes/GameOfRunes/Models/Stage.swift:31]: num waves should be hardcoded i guess, we can add the array of enemies here, and persist this entire class?

File: GameOfRunes/GameOfRunes/Models/Stage.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//  GameOfRunes
+//
+//  Created by Jermy on 29/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+import UIKit
+
+struct Stage {
+    enum Category: String {
```
Comment (dongsiji) [inline GameOfRunes/GameOfRunes/Models/Stage.swift:12]: do we need a category? Shouldn't every stage have a boss?

File: GameOfRunes/GameOfRunes/Models/Stage.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    enum AchievementLevel: String {
+        case S
+        case A
+        case B
+        case C
+        case empty
+    }
+    
+    let name: String
+    let groupName: String
```
Comment (dongsiji) [inline GameOfRunes/GameOfRunes/Models/Stage.swift]: what is groupName used for?

File: GameOfRunes/GameOfRunes/Types/Avatar.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
+//
+//  Avatar.swift
+//  GameOfRunes
+//
+//  Created by Jermy on 29/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+enum Avatar: Int, CaseIterable {
```
Comment (dongsiji) [inline GameOfRunes/GameOfRunes/Types/Avatar.swift:9]: looks gud

**Review by dongsiji — APPROVED** [review]
lgtm 👍

**Review by brianyenna — APPROVED** [review]
LGTM. Thanks for the crazy hard work on the front end man :)

File: GameOfRunes/GameOfRunes/Models/Stage.swift (hunk 1 above)
Comment (JermyTan) [reply GameOfRunes/GameOfRunes/Models/Stage.swift:31]: Yep, that's the plan

Comment (JermyTan) [reply GameOfRunes/GameOfRunes/Models/Stage.swift:31]: So the `numWaves` value here will dictate the spawn engine's behaviour?

File: GameOfRunes/GameOfRunes/Models/Stage.swift (hunk 2 above)
Comment (JermyTan) [reply GameOfRunes/GameOfRunes/Models/Stage.swift:12]: My idea is sth like a "boss" stage to mark the end of a "Chapter". Each Chapter represents a series of stages in a region of the map - can think of it like the different worlds in plants vs zombies. What do the rest think? @andyylam @brianyenna

File: GameOfRunes/GameOfRunes/Models/Stage.swift (hunk 3 above)
Comment (JermyTan) [reply GameOfRunes/GameOfRunes/Models/Stage.swift]: Oh right, I should rename `groupName` to `chapter`.

**Pull request merged**